bazel/linux/defs.bzl

```diff
@@ -311,3 +311,23 @@ Example:
         ),
     },
 )
+
+def kernel_unit_test(name, module, linux_image, rootfs_image):
+    """Convenience wrapper around sh_test, makes it easier to declare and run KUnit user-mode linux tests.
+
+    Args:
+      name: string, name for the underlying sh_test rule.
+      module: label, target KUnit kernel module containing the tests.
+      linux_image: label, executable user-mode linux image file.
+      rootfs_image: label, rootfs image file to be used by the linux image.
+    """
+    srcs = ["run_um_kunit_tests.sh"]
```

Comment:
What's the best way to expose/declare this file?

Ideally, we want store this here because it is common to all kernel unit tests. How would this interact with other repositories importing this definition from their WORKSPACE file?

Reply:
After a bit of digging it seems that the `export_files` built-in function does exactly that (https://docs.bazel.build/versions/master/be/functions.html#exports_files). I'll test its behavior and then update the PR.

Reply:
Now the macro correctly references the local run_um_kunit_tests.sh script. This requires a new parameter representing the name assigned to this project when imported through the WORKSPACE file. I don't know if this is common practice/acceptable, I'm open to other suggestions on how to handle this.

Reply:
For rules, it just works? See for example the `_validator` or `_expander` use in `bazel/enkit/defs.bzl`, they use the path relative to the enkit WORKSPACE, despite being referenced from the internal WORKSPACE.

For macros, I suspect you are right, as the macro expansion happens in the context of the including WORKSPACE. I've asked for the best way on the bazel slack channel, but no reply yet. A bunch of stackoverflow questions talk about how to retrieve the name of the repository/workspace, but not sure they work / are correct, they talk about ctx which is not accessible from a macro.

I would leave it as is, modulo the comment above.